pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -70,8 +71,31 @@ func getAndPublishMetrics(ctx *domainContext, hyper hypervisor.Hypervisor) {
 			// We clear the memory so it doesn't accidentally get
 			// reported.  We keep the CPUTotal and AvailableMemory
 			dm.UsedMemory = 0
+			dm.MaxUsedMemory = 0
 			dm.UsedMemoryPercent = 0
 		}
+
+		config := lookupDomainConfig(ctx, status.Key())
+		if config == nil {
+			continue
+		}
+
+		var CurrMaxUsedMemory uint32
+		st, _ := ctx.pubDomainMetric.Get(dm.Key())
+		if st != nil {
+			previous_metric := st.(types.DomainMetric)
+			CurrMaxUsedMemory = previous_metric.MaxUsedMemory
+		}
+
+		if CurrMaxUsedMemory < dm.MaxUsedMemory {
+			used_pct := dm.MaxUsedMemory * 100 * 1024 / uint32(config.Memory)
+			logrus.Infof("Memory watermark for %s increased: %d MiB,"+
```

Comment:
Use log.Noticef() instead,

Reply:
It's unclear to me, why are we using both `log` and `logrus`? And why in this particular case `log.Noticef` is a better choice?

Reply:
Because it provides source annotation for filtering